### PR TITLE
Review for the Paxos blog post

### PR DESCRIPTION
**Pull request by gvsmirnov**
All in all, quite straightforward, I like this!
There are some inline comments, I'll transfer them to github comments momentarily.

### CODE REVIEW

File: _posts/2016-05-01-paxos.html [hunk 1]
```diff
@@ -18,47 +18,47 @@
...
 
   <ul>
-    <li>They do. For example <a class="link" href="http://treode.github.io/">TreodeDB</a> used this approach. It's interesting that I haven't seen any academic paper dedicated to this idea, but I saw several industry systems built with it. So I practiced my math fu, described and proved FSM over Single Decree Paxos in the <a class="link" href="http://rystsov.info/2015/09/16/how-paxos-works.html">How Paxos works</a> post.</li>
+      <li>They do exist. For example, <a class="link" href="http://treode.github.io/">TreodeDB</a> used this approach. It's interesting that I haven't seen any academic paper dedicated to this <!-- define this --> idea, but I saw several industry systems built with it. So I practiced my Math-Fu, described and proved FSM over Single Decree Paxos in the <a class="link" href="http://rystsov.info/2015/09/16/how-paxos-works.html">How Paxos Works</a> post.</li>
```
Comment (gvsmirnov) [inline _posts/2016-05-01-paxos.html]: > seen any academic paper dedicated to this

Maybe, define `this`?

File: _posts/2016-05-01-paxos.html [hunk 2]
```diff
@@ -18,47 +18,47 @@
...
+      <li>IMHO the log-based solutions are more popular because they had the advantage of time. Until Vertical Paxos was published in <span class="num">2009</span>, log replication was the only consensus protocol capable of cluster membership change. Thus, it was the only practical solution for highly available services. But now we have a few more algorithms to choose from. For example Vertical Paxos and Raft's joint consensus membership change are among them. Take a look at the <a class="link" href="http://rystsov.info/2016/01/05/raft-paxos.html">Best of both worlds: Raft's joint consensus + Single Decree Paxos</a> post if you're interested in understanding how dynamic membership works with a distributed variable <!-- distributed variable? Either I'm missing something, or it's not clear what this means in this context -->.</li>
```
Comment (gvsmirnov) [inline _posts/2016-05-01-paxos.html]: > how dynamic membership works with a distributed variable

Not clear to me what this means. Works with a distributed variable?

File: _posts/2016-05-01-paxos.html [hunk 3]
```diff
@@ -79,13 +79,15 @@
 
   <h5>Distributed FSM</h5>
 
-  <p>Distributed FSM has several endpoints with the execute service, but it doesn't matter which endpoint you choose because the system behaves exactly as a system with several endpoints but leading to the same (non-distributed) FSM. The only difference that the distributed version runs on <span class="num">2n+1</span> nodes and tolerates up to <span class="num">n</span> failures.</p>
+  <p>Distributed FSM has several endpoints <!-- define endpoints --> with the <code>execute</code> service, but it doesn't matter which endpoint you choose because the system behaves exactly like a system with several endpoints but leading to the same (non-distributed) FSM. <!-- refactor this whole sentence, perhaps split it to 2 or more smaller sentences --> The only difference is that the distributed version runs on <span class="num">2n+1</span> nodes and tolerates up to <span class="num">n</span> failures.</p>
```
Comment (gvsmirnov) [inline _posts/2016-05-01-paxos.html]: Define `endpoint`?

File: _posts/2016-05-01-paxos.html [hunk 4]
```diff
@@ -79,13 +79,15 @@
...
```
Comment (gvsmirnov) [inline _posts/2016-05-01-paxos.html:83]: In general, this paragraph is not very understandable, especially the first sentence needs refactoring.

File: _posts/2016-05-01-paxos.html [hunk 5]
```diff
@@ -79,13 +79,15 @@
...
 
   <h5>Visualization</h5>
 
-  <p>The visualization consists of two code columns and the data side view. The first column contains code of a distributed Single Decree Paxos FSM engine (36 lines of code); while the second column is dedicated for the state transition functions and the generals behavior (33 lines of code).</p>
+  <p>The visualization consists of two code columns and a side view with data. The first column contains the code of a distributed Single Decree Paxos FSM engine (36 lines of code); while the second column is dedicated for the state transition functions and the generals behavior (33 lines of code).</p>
 
-  <p>It’s worth to mention that the state transition functions weren't changed at all, so it extremely easy to convert an existing FSM into a DFSM. By the way the actions in the example look like message handlers in the actor model, so FSM over SDP looks like the great way to store the state of the actors from Akka or Orleans frameworks.</p>
```
Comment (gvsmirnov) [inline _posts/2016-05-01-paxos.html:88]: Maybe better put the visualization right here?

File: _posts/2016-05-01-paxos.html (hunk 2 above)
Comment (rystsov) [inline _posts/2016-05-01-paxos.html]: You're right. I forgot to define it first, going to fix.

File: _posts/2016-05-01-paxos.html (hunk 4 above)
Comment (rystsov) [inline _posts/2016-05-01-paxos.html:83]: You're right I should work on the end of the post.

**Pull request merged**